File: packages/shared-types/src/insight.ts
export type InsightSeverity = "CRITICAL" | "WARNING" | "INFO"

export interface InsightCard {
  id: string
  signalId: string
  severity: InsightSeverity
  title: string
  what: string
  why: string
  evidence: string[]
  confidence: number
  agent: string
  createdAt: string
  dismissedAt?: string | null
  snoozedUntil?: string | null
}

export interface InsightOutcome {
  id: string
  insightId: string
  measuredAt: string
  windowHours: 24 | 48 | 168
  metricDeltas: {
    roas_delta?: number
    spend_delta?: number
    revenue_delta?: number
    cac_delta?: number
    [key: string]: number | undefined
  }
  outcomeScore: number
}

File: services/orchestrator/src/schemas/insight.py
"""InsightCard schema."""

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


class InsightCard(BaseModel):
    id: str
    signal_id: str
    severity: Literal["critical", "warning", "info"]
    title: str = Field(max_length=80)
    what: str
    why: str
    evidence: list[str]
    confidence: float = Field(ge=0.0, le=1.0)
    agent: str = "insight_agent"
    created_at: datetime
